vault/eventbus/bus.go

```diff
@@ -47,14 +58,27 @@ func (bus *EventBus) Start() {
 	}
 }
 
-// Send sends an event to the event bus and routes it to all relevant subscribers.
+// SendInternal sends an event to the event bus and routes it to all relevant subscribers.
 // This function does *not* wait for all subscribers to acknowledge before returning.
-func (bus *EventBus) Send(ctx context.Context, eventType logical.EventType, data *logical.EventData) error {
+// This function is meant to be used by trusted internal code, so it can specify details like the namespace
+// and plugin info. Events from plugins should be routed through WithPlugin(), which will populate
+// the namespace and plugin info automatically.
+func (bus *EventBus) SendInternal(ctx context.Context, namespace *namespace.Namespace, pluginInfo *logical.EventPluginInfo, eventType logical.EventType, data *logical.EventData) error {
 	if !bus.started.Load() {
 		return ErrNotStarted
 	}
-	bus.logger.Info("Sending event", "event", data)
-	_, err := bus.broker.Send(ctx, eventlogger.EventType(eventType), data)
+	var nspace string
+	if namespace != nil {
```

Comment:
Similar to below, I think we should probably error on nil namespace, I can't think of any valid reason to omit it.